spec/derived_build_code_formatter_spec.rb

```diff
@@ -81,6 +74,125 @@
     end
   end
 
+  describe 'configurable digit counts' do
+    context 'with custom digit counts' do
+      it 'uses 1 digit for each component' do
+        version = Fastlane::Models::AppVersion.new(1, 2, 3, 4)
+        formatter = described_class.new(prefix: '1', major_digits: 1, minor_digits: 1, patch_digits: 1, build_digits: 1)
+        build_code_string = formatter.build_code(version: version)
+        expect(build_code_string.to_s).to eq('11234')
+      end
+
+      it 'uses 2 digits for each component and pads with zeros' do
+        # Test both large numbers and zero-padding in one test
+        version_large = Fastlane::Models::AppVersion.new(12, 34, 56, 78)
+        formatter = described_class.new(prefix: '2', major_digits: 2, minor_digits: 2, patch_digits: 2, build_digits: 2)
+        expect(formatter.build_code(version: version_large).to_s).to eq('212345678')
+
+        # Test zero-padding with smaller numbers
+        version_small = Fastlane::Models::AppVersion.new(1, 2, 3, 4)
+        expect(formatter.build_code(version: version_small).to_s).to eq('201020304')
+      end
+
+      it 'uses mixed digit counts' do
+        version = Fastlane::Models::AppVersion.new(1, 23, 45, 678)
+        formatter = described_class.new(prefix: '', major_digits: 1, minor_digits: 2, patch_digits: 2, build_digits: 3)
+        build_code_string = formatter.build_code(version: version)
+        # 1(1 digit) + 23(2 digits) + 45(2 digits) + 678(3 digits) = "12345678"
+        expect(build_code_string.to_s).to eq('12345678')
+      end
+
+      it 'handles maximum values within digit limits' do
+        version = Fastlane::Models::AppVersion.new(99, 99, 99, 99)
+        formatter = described_class.new(prefix: '1', major_digits: 2, minor_digits: 2, patch_digits: 2, build_digits: 2)
+        build_code_string = formatter.build_code(version: version)
+        expect(build_code_string.to_s).to eq('199999999')
+      end
+    end
+
+    context 'with empty prefix and custom digits' do
+      it 'trims leading zeros correctly with 1-digit major' do
+        version = Fastlane::Models::AppVersion.new(5, 12, 34, 56)
+        formatter = described_class.new(prefix: '', major_digits: 1, minor_digits: 2, patch_digits: 2, build_digits: 2)
+        build_code_string = formatter.build_code(version: version)
+        expect(build_code_string.to_s).to eq('5123456')
+      end
+
+      it 'trims leading zeros correctly with larger major' do
```

Comment:
💄  The name of this test felt a bit misleading to me, as you're mentioning testing with "larger major", but the major is just `7` (sure, `7` is larger than the `5` used in the previous test… but was confused as to why that would be important).

It took me a minute to finally understand that this was about a larger _number of major digits_. So might be worth updating the test title to make this clearer 😅 

```suggestion
      it 'trims leading zeros correctly with larger number of major digits' do
```